openedx/core/lib/api/paginators.py

```diff
@@ -29,6 +30,38 @@ def get_paginated_response(self, data):
         })
 
 
+class NamespacedPageNumberPagination(pagination.PageNumberPagination):
+    """
+    Pagination scheme that returns results with pagination metadata
+    embedded in a "pagination" attribute.  Can be used with data
+    that comes as a list of items, or as a dict with a "results"
+    attribute that contains the code.
+    """
+
+    page_size_query_param = "page_size"
+
+    def get_paginated_response(self, data):
+        """
+        Annotate the response with pagination information
+        """
+        metadata = OrderedDict([
```

Comment:
See comment above.  In short, it's what DRF does.  It gives a nicer response to the end-user.